packages/builtin/array.pony

```diff
@@ -63,11 +63,11 @@ class Array[A] is Seq[A]
     """
     _ptr._offset(from_offset)._copy_to(ptr._offset(to_offset), copy_len)
 
-  fun cpointer(): Pointer[A] tag =>
+  fun cpointer(offset: USize = 0): Pointer[A] tag =>
     """
     Return the underlying C-style pointer.
     """
-    _ptr
+    _ptr._offset(offset)
```

Comment:
Does this change allow Pony programs to break memory safety (without using FFI)?  After thinking it through, I don't think it does, but I wanted to raise the point to ask, because `Pointer._offset` is private.